superset/assets/javascripts/components/FilterableTable/FilterableTable.jsx

```diff
@@ -0,0 +1,190 @@
+import { List } from 'immutable';
+import React, { PropTypes, PureComponent } from 'react';
+import {
+  Column,
+  Table,
+  SortDirection,
+  SortIndicator,
+} from 'react-virtualized';
+import { getTextWidth } from '../../modules/visUtils';
+
+const propTypes = {
+  orderedColumnKeys: PropTypes.array.isRequired,
+  data: PropTypes.array.isRequired,
+  height: PropTypes.number.isRequired,
+  filterText: PropTypes.string,
+  headerHeight: PropTypes.number,
+  overscanRowCount: PropTypes.number,
+  rowHeight: PropTypes.number,
+  striped: PropTypes.bool,
+};
+
+const defaultProps = {
+  filterText: '',
+  headerHeight: 32,
+  overscanRowCount: 10,
+  rowHeight: 32,
+  striped: true,
+};
+
+export default class FilterableTable extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.list = List(this.formatTableData(props.data));
+    this.headerRenderer = this.headerRenderer.bind(this);
+    this.rowClassName = this.rowClassName.bind(this);
+    this.sort = this.sort.bind(this);
+
+    this.widthsForColumnsByKey = this.getWidthsForColumns();
+    this.totalTableWidth = props.orderedColumnKeys
+      .map(key => this.widthsForColumnsByKey[key])
+      .reduce((curr, next) => curr + next);
+
+    this.state = {
+      sortBy: props.orderedColumnKeys[0],
+      sortDirection: SortDirection.ASC,
+      fitted: false,
+    };
+  }
+
+  hasMatch(text, row) {
+    const values = [];
+    for (const key in row) {
+      if (row.hasOwnProperty(key)) {
+        values.push(row[key].toLowerCase());
+      }
+    }
+    return values.some(v => v.includes(text.toLowerCase()));
+  }
+
+  formatTableData(data) {
+    const formattedData = data.map((row) => {
+      const newRow = {};
+      for (const k in row) {
+        const val = row[k];
+        if (typeof(val) === 'string') {
+          newRow[k] = val;
+        } else {
+          newRow[k] = JSON.stringify(val);
+        }
+      }
+      return newRow;
+    });
+    return formattedData;
+  }
+
+  getWidthsForColumns() {
```

Comment:
I'm concerned about the potential cost of this method. `getTextWidth` creating a canvas object for each cell in the dataset seems costly (though I haven't researched it..). Should be use a sample of rows here?